### PR TITLE
chore: label SyncSetInstances with (Selector)SyncSetRef

### PR DESCRIPTION
**Pull request by djzager**
Add a label to SyncSetInstances with the SelectorSyncSet or SyncSet they
are associated with. This makes it possible, knowing the (Selector)SyncSet, to find SyncSetInstances that were created.

### CODE REVIEW

File: pkg/controller/syncset/syncset_controller.go [hunk 1]
```diff
@@ -396,6 +396,9 @@ func (r *ReconcileSyncSet) syncSetInstanceForSyncSet(cd *hivev1.ClusterDeploymen
 			Name:            syncSetInstanceNameForSyncSet(cd, syncSet),
 			Namespace:       cd.Namespace,
 			OwnerReferences: []metav1.OwnerReference{*cdRef},
+			Labels: map[string]string{
+				"SyncSetName": syncSet.Name,
```
Comment (staebler) [inline pkg/controller/syncset/syncset_controller.go]: Use the label key of `hive.openshift.io/sync-set-name`.

File: pkg/controller/syncset/syncset_controller.go [hunk 2]
```diff
@@ -421,6 +424,9 @@ func (r *ReconcileSyncSet) syncSetInstanceForSelectorSyncSet(cd *hivev1.ClusterD
 			Name:            syncSetInstanceNameForSelectorSyncSet(cd, selectorSyncSet),
 			Namespace:       cd.Namespace,
 			OwnerReferences: []metav1.OwnerReference{*cdRef},
+			Labels: map[string]string{
+				"SelectorSyncSetName": selectorSyncSet.Name,
```
Comment (staebler) [inline pkg/controller/syncset/syncset_controller.go]: Use the label key of `hive.openshift.io/selector-sync-set-name`.

**Comment by twiest** [comment]
Yeah, I agree. Thanks @djzager.

I added a [PR](https://github.com/openshift/hive/pull/794) that adds similar functionality plus cleans up a few other issues with the labels.

Does that PR address what you were looking for?

**Pull request closed without merging**